nucypher/characters/control/specifications/fields/file.py

```diff
@@ -0,0 +1,33 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import os
+from marshmallow import fields
+
+from nucypher.characters.control.specifications.exceptions import InvalidInputData, InvalidNativeDataTypes
+from nucypher.characters.control.specifications.fields.base import BaseField
+
+
+class FileField(BaseField, fields.String):
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        with open(value, 'rb') as plaintext_file:
+            plaintext = plaintext_file.read()
```

Comment:
do we need to be concerned about file size here i.e. what about a really large file? Would something buffered be more prudent?

Reply:
It's going to have to be in memory eventually... not sure.

Reply:
oh, unless we want to put each buffer in its own message kit and output them individually?

Reply:
but yeah, we could put a size check into the validation of the FileField

Reply:
> oh, unless we want to put each buffer in its own message kit and output them individually?

The underlying pyumbral `pre.encrypt(...)` takes bytes, so something like chunking the file data and each chunk producing its own message kit would be a solution. We did chat some time ago about Enrico being able to handle a stream of data.

Reply:
Hmm, time to rename this PR ;-).  Le't kick out the large file handling to a follow up PR.

Reply:
Needs tracking this with a TODO/issue

Reply:
#2106